syntax = "proto3";

package client_api_proto;

service CryptoAttacksService {
  rpc getAvailableServices(EmptyMessage) returns (ExpectedValues) {};
}

// TODO: move services to the separate module
service DigitalSignatureAttackService {
  rpc ecdsaReusedNonceAttack(ReusedNonceAttackRequest) returns (ReusedNonceAttackResponse) {};
}

message MapFieldEntry {
  optional string key = 1;
  optional string value = 2;
}

message ExpectedValues {
  repeated MapFieldEntry entries = 1;
}

message ReusedNonceAttackRequest {
  optional bytes pubkey_order = 1;
  optional bytes signature1 = 2;
  optional bytes signature2 = 3;
  optional bytes msg_hash1 = 4;
  optional bytes msg_hash2 = 5;
}

message ReusedNonceAttackResponse {
  optional bytes private_key = 1;
}

message EmptyMessage {

}
